File: backend_ddd/app/Models/Bus.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Bus extends Model
{
    protected $fillable = ['number', 'line_name'];

    public function stops(): BelongsToMany
    {
        return $this->belongsToMany(Stop::class, 'stop_bus');
    }

    public function locations(): HasMany
    {
        return $this->hasMany(BusLocation::class);
    }

    public function currentLocation(): HasOne
    {
        return $this->hasOne(BusLocation::class)->latestOfMany();
    }
}

File: front_ddd/lib/features/buses/domain/entities/bus.dart
import 'package:equatable/equatable.dart';
import 'bus_location.dart';

class Bus extends Equatable {
  final int id;
  final String number;
  final String lineName;
  final BusLocation? currentLocation;

  const Bus({
    required this.id,
    required this.number,
    required this.lineName,
    this.currentLocation,
  });

  @override
  List<Object?> get props => [id, number, lineName, currentLocation];
}
